CMakeLists.txt

```diff
@@ -531,6 +531,7 @@ endforeach(def_name)
 genex_for_option(ASAN)
 genex_for_option(UBSAN)
 foreach(target devilution ${BIN_TARGET} SourceS)
+target_link_libraries(${target} PRIVATE stdc++ pspgum SDL2_ttf freetype SDL2_image png jpeg SDL2_mixer vorbisfile vorbis ogg psphprm mikmod pspaudiocodec pspaudio SDL2main SDL2 g c GL GLU glut z m pspvfpu pspvram  -L${PSPDEV}/psp/sdk/lib -L${PSPDEV}/psp/lib pspdebug pspgu pspctrl pspge pspirkeyb psppower psppower_driver pspdisplay psphprm pspsdk psprtc psputility pspnet_inet  c pspuser)

```

Comment:
linker order matters
e.g. you have to link `c` last

move this where the `if(SWITCH)` is

Reply:
most of the work was spent getting this in an order that would even link :D

Reply:
At least, using the PSP sdk in proper, you'd have a makefile which would make it a bit more readable then having to cram it all on one line. You usually start with static libraries, then add the stubs for the kernel imports and/or sdk modules.
```
LDLIBS		= -lgum -lgu -lwave -lm
LOADLIBES	=  $(LIBDIR)/ctrl_stub.a
LOADLIBES	+= $(LIBDIR)/display_stub.a
LOADLIBES	+= $(LIBDIR)/ge_user_stub.a

TARGET	= balloon
OBJS	= main.o balloon.o cloud.o sphere.o pole.o

all:	$(TARGET).elf
```